### DISCUSSION TITLE
How to use OSC Messages

### DISCUSSION THREAD

**Discussion by Wapata** [alf45tar/PedalinoMini#463] [Q&A] [answered]
Hi !
i have a working PedalinoMini in a working Wifi environment and a working QmidiNet.
But now that I want to use OSC, it seem that I don't know how to use it.
So, PedalinoMini is sending midi stuff to my Zynthian box, Zynthian use OSC this way : https://wiki.zynthian.org/index.php/Accessing_Zynthian_from_your_computer#OSC

 I've tried to send an OSC message via button one this way: 
OSC Adress
zynthian.local:1370/CUIA/SCREEN_MAIN

![Capture du 2023-09-17 15-44-27](https://github.com/alf45tar/PedalinoMini/assets/15179434/edf8849e-5f47-4edf-bcb9-2c2c2ca9f490)

And many other variant but nothing reach the Zynthian and I don't know how to send OSC message via other tools neither.
So.. What am I doing wrong ?

See ya !

**Comment by alf45tar** [comment, marked as answer]
Set your Zynthian box name or IP and port under Options->OSC Remote

**Comment by Wapata** [comment]
Ooooooh.
That mean that we can't control two devices !
Well, that not the main issue here haha.

Anyway it doesn't work on my side, i put the zynthian (without ".local") and then the IP, and the reboot and test that others midi message are working well.... But don't see the OSC working

Please note than the OSC Adress is now /CUIA/SCREEN_MAIN (or CUIA/SCREEN_MAIN)

**Comment by alf45tar** [comment]
To control multiple devices you had to use broadcast address 255.255.255.255

**Comment by Wapata** [comment]
Oh ! IT WORK !
Only for /CUIA/ARROW_RIGHT message for now, but it work.
Your first answer was a perfect match, i will go to zynthian side to see what's wrong now !

**Comment by Wapata** [comment]
255.255.255.255
IP Adrees (192.blabla)
zynthian (without local)

Everything work as soon as the Port (1370 for my zynthian) match.

**Comment by Wapata** [comment]
Little note for people in the future : take care of the osc list of your receiver. The one of Zynthian have evolved and the first command I tried to use was not usable anymore but still in the example section of the project. For Zynthian documentation it's changed now... but take care of that ;-)
